scripts/testing_scripts/test_all_sandia

```diff
@@ -508,10 +497,7 @@ elif [ "$MACHINE" = "apollo" ]; then
 
   if [ "$SPOT_CHECK" = "True" ]; then
     # Format: (compiler module-list build-list exe-name warning-flag)
-    COMPILERS=("gcc/4.8.4 $BASE_MODULE_LIST "OpenMP,Pthread" g++ $GCC_WARNING_FLAGS"
-               "gcc/5.3.0 $BASE_MODULE_LIST "Serial" g++ $GCC_WARNING_FLAGS"
-               "intel/16.0.1 $BASE_MODULE_LIST "OpenMP" icpc $INTEL_WARNING_FLAGS"
-               "clang/3.9.0 $BASE_MODULE_LIST "Pthread_Serial" clang++ $CLANG_WARNING_FLAGS"
+    COMPILERS=("gcc/5.3.0 $BASE_MODULE_LIST "OpenMP,Pthread,Serial" g++ $GCC_WARNING_FLAGS"
```

Comment:
in updating the read me, I saw that we removed selective version of clang...on apollo, clang3.9.0 uses gcc 5.3.0 and clang says that c++14 is supported with 3.4 or later.  Thus, I'm not sure if we were supposed to remove clang 3.9.0

Reply:
Good catch.  Looking at the [developer meeting notes on the internal repo](https://github.com/kokkos/internal-documents/blob/master/meeting-notes/2020-08-12.md) we decided we'd drop support for the 3.x series.  So Nathan's changes are right.